aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,336 @@
+
+Use the ``IHtmlLocalizer`` implementation for resources that contain markup. ``IHtmlLocalizer`` HTML encodes arguments that are formatted in the resource string, but not the resource string. In the sample highlighted below, only the ``name`` parameter is HTML encoded.
+
+.. literalinclude:: localization/sample/Controllers/BookController.cs
+  :language: c#
+  :linenos:
+  :lines: 1-23
+  :emphasize-lines: 3,5,20 
+
+At the lowest level, you can get ``IStringLocalizerFactory`` out of :doc:`/fundamentals/dependency-injection`:
+
+.. literalinclude:: localization/sample/Controllers/TestController.cs
+  :language: c#
+  :linenos:
+  :lines: 9-27
+  :emphasize-lines: 6-11 
+  :dedent: 3
+
+The code above demonstrates each of the two factory create methods. 
+
+You can partition your localized strings by controller, area, or have just one container. In the sample app, a dummy class named ``SharedResource`` is used to contain shared resources. Some developers use the ``Startup`` class to contain global or shared strings.  In the sample below, the ``ZebraController`` and the ``SharedResource`` localizers are used:
+
+.. literalinclude:: localization/sample/Controllers/ZebraController.cs
+  :language: c#
+  :linenos:
+  :lines: 9-27
+  :dedent: 3
+
+View localization
+--------------------
+
+The ``IViewLocalizer`` service provides localized strings for a `view <http://docs.asp.net/projects/mvc/en/latest/views/index.html>`_. The ``ViewLocalizer`` class implements this interface and finds the resource location from the view file path. The following code shows how to use the default implementation of ``IViewLocalizer``:
+
+.. literalinclude:: localization/sample/Views/Home/About.cshtml
+  :language: HTML
+  
+The default implementation of ``IViewLocalizer`` finds the resource file based on the view name. There is no option to use a global shared resource file. ``ViewLocalizer`` implements the localizer using ``IHtmlLocalizer``, so Razor doesn't HTML encode the localized string. You can parameterize resource strings and ``IViewLocalizer`` will HTML encode the parameters, but not the resource string. Consider the following Razor markup:
+
+.. code-block:: HTML
+
+  @Localizer["<i>Hello</i> <b>{0}!</b>", UserManager.GetUserName(User)]
```

Comment:
I see the use of this section, but perhaps we might have a warning that in general it's better only localize text, and not HTML? Certainly there's use cases for localizing the html itself but it's a bit more dangerous so a warning might save them some pain.